### ISSUE TITLE
Missing npm package for `@typescript-eslint/consistent-type-definitions`?

### ISSUE THREAD

**Issue by hellatan** [typescript-eslint/typescript-eslint#753]
I was trying to install `@typescript-eslint/consistent-type-definitions` in my repo but I end up with the following error:
```
error An unexpected error occurred: "http://registry.npmjs.org/@typescript-eslint%2fconsistent-type-definitions: Not found".
```
I went onto npmjs.com looking for this package as well but get 0 results:

<img width="1552" alt="Screen Shot 2019-07-24 at 11 21 53 AM" src="https://user-images.githubusercontent.com/2065840/61806184-44920300-ae05-11e9-890a-a3121458e912.png">

Even if i remove the `@typescript-eslint` namespace, there are still 0 results:

<img width="1552" alt="Screen Shot 2019-07-24 at 11 24 21 AM" src="https://user-images.githubusercontent.com/2065840/61806413-9b97d800-ae05-11e9-85b3-ee2f30d7e9d8.png">

Searching for `@typescript-eslint/parser` returns the npm page itself. 

Any thoughts on what I may be missing? Am I using the incorrect namespace or something?

**Comment by bradzacher** [comment]
Please have a read of the plugin's readme. There are instructions for installation and usage there.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/README.md

**Comment by hellatan** [comment]
@bradzacher just to be crystal clear, all the rules are bundled up when installing the specific version of the `@typescript-eslint/eslint-plugin` (accompanied with the same version of the `@typescript-eslint/parser`) is what the README is indicating? To me, it doesn't really indicate that very clearly. I can update the README if that is the case.

**Comment by bradzacher** [comment]
I mean, I don't know how much clearer you can get than the first section of the readme, the installation instructions?

> ## Installation
> 
> Make sure you have TypeScript and @typescript-eslint/parser installed, then install the plugin:
> 
> ```sh
> npm i @typescript-eslint/eslint-plugin --save-dev
> ```

This pretty clearly indicates that exactly one package is published for the eslint plugin?
Additionally - every single eslint plugin works this way; one package is published with all of the rules, not one package per rule.

The problem is that you're conflating eslint's rule configuration syntax with NPM scoped packages. Whilst they look the same, they are not the same (`@typescript-eslint/consistent-type-definitions` looks like a scoped package, but it is actually a shorthand that eslint resolves to `@typescript-eslint/eslint-plugin/consistent-type-definitions`).

Similarly the config `{ "plugins": ["@typescript-eslint"] }` is shorthand for `{ "plugins": ["@typescript-eslint/eslint-plugin"] }`

More information about eslint and configuring rules can be found in the eslint docs https://eslint.org/docs/user-guide/configuring

**Comment by hellatan** [comment]
yes, you're right - i am conflating NPM's installation via namespacing and eslint's automagical path resolutions (something I've always hated), that's why I was getting confused. I don't think it's too crazy for the confusion to happen, though, based on the config syntax. I always forget about that when I have to go update one of our configs that I haven't updated in a while. Thanks for clearing things up!
